src/responseType.ts

```diff
@@ -0,0 +1,24 @@
+interface BaseResponse {
```

Comment:
I would really like to avoid writing custom types for fetch and stick to the build in typescript ones. We gonna miss a lot of handing by doing this that is very likely to be covered. We are here ignoring all error cases and expecting that the only error that may ever happen is an internal API error successfully returned to the client.

Reply:
We wouldn't get typing for the body if we are not using a custom type which we want to have. However, what we could do it extend from the normal Response type like this:
```typescript
interface SuccessResponse<Body> extends Omit<Response, 'body'> {
  ok: true;
  status: 200 | 201 | 204;
  body: Body;
}
```